### PR TITLE
Remove Read/WriteProcessMemory from PAL.

### PR DESCRIPTION
**Pull request by mikem8361**
Ifdef more unused code that uses ReadProcessMemory. Move the current
memory probing in the transport to PAL_ProbeMemory. Add PAL_ProbeMemory
to dac PAL exports.

PAL_ProbeMemory may be changed to use write/read on a pipe to
validate the memory as soon as we make it perform as well as
the current code.

### CODE REVIEW

File: src/pal/tests/palsuite/filemapping_memmgt/CMakeLists.txt [hunk 1]
```diff
@@ -17,7 +17,6 @@ add_subdirectory(LockFile)
 add_subdirectory(MapViewOfFile)
 add_subdirectory(OpenFileMappingA)
 add_subdirectory(OpenFileMappingW)
-add_subdirectory(ReadProcessMemory)
```
Comment (jkotas) [inline src/pal/tests/palsuite/filemapping_memmgt/CMakeLists.txt]: Physically delete these tests as well?

**Comment by jkotas** [comment]
Now that this is PAL API, it may be nice to add PAL test to verify that it actually works as expected.

**Comment by jkotas** [comment]
LGTM otherwise

**Comment by janvorli** [comment]
LGTM. I would also like to have a PAL test for that.

File: src/jit/ee_il_dll.cpp [hunk 2]
```diff
@@ -1537,6 +1537,9 @@ const char* Compiler::eeGetClassName(CORINFO_CLASS_HANDLE clsHnd)
 
 const wchar_t* Compiler::eeGetCPString(size_t strHandle)
 {
+#ifdef FEATURE_PAL
```
Comment (janvorli) [inline src/jit/ee_il_dll.cpp]: @BruceForstall what do you think about this change?

File: src/pal/tests/palsuite/filemapping_memmgt/CMakeLists.txt (hunk 1 above)
Comment (mikem8361) [inline src/pal/tests/palsuite/filemapping_memmgt/CMakeLists.txt]: Removed ReadProcessMemory and added PAL_ProbeMemory tests.

**Pull request merged**